packages/@aws-cdk/aws-ecs/test/container-definition.test.ts

```diff
@@ -997,6 +997,132 @@ describe('container definition', () => {
+        memoryLimitMiB: 1024,
+        inferenceAcceleratorResources,
+      });
+
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+        Family: 'Ec2TaskDef',
+        InferenceAccelerators: [{
+          DeviceName: 'device1',
+          DeviceType: 'eia2.medium',
+        }],
+        ContainerDefinitions: [
+          {
+            Image: 'test',
+            ResourceRequirements: [
+              {
+                Type: 'InferenceAccelerator',
+                Value: 'device1',
+              },
+            ],
+          },
+        ],
+      });
+
+
+    });
+    test('correctly adds inference accelerator resources to container definition using both props and addInferenceAcceleratorResource method', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      const inferenceAccelerators = [{
+        deviceName: 'device1',
+        deviceType: 'eia2.medium',
+      }, {
+        deviceName: 'device2',
+        deviceType: 'eia2.large',
+      }];
+
+      const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDef', {
+        inferenceAccelerators,
+      });
+
+      const inferenceAcceleratorResources = ['device1'];
+
+      const container = taskDefinition.addContainer('cont', {
+        image: ecs.ContainerImage.fromRegistry('test'),
+        memoryLimitMiB: 1024,
+        inferenceAcceleratorResources,
+      });
+
+      // WHEN
+      container.addInferenceAcceleratorResource('device2');
+
+      // THEN
+      expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+        Family: 'Ec2TaskDef',
+        InferenceAccelerators: [{
+          DeviceName: 'device1',
+          DeviceType: 'eia2.medium',
+        }, {
+          DeviceName: 'device2',
+          DeviceType: 'eia2.large',
+        }],
+        ContainerDefinitions: [
+          {
+            Image: 'test',
+            ResourceRequirements: [
+              {
+                Type: 'InferenceAccelerator',
+                Value: 'device1',
+              },
+              {
+                Type: 'InferenceAccelerator',
+                Value: 'device2',
+              },
+            ],
+          },
+        ],
+      });
+
+    });
+    test('throws when the value in resource requirement does not match any inference accelerators defined in the Task Definition', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      const inferenceAccelerators = [{
+        deviceName: 'device1',
+        deviceType: 'eia2.medium',
+      }];
+
+      const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDef', {
+        inferenceAccelerators,
+      });
+
+      const inferenceAcceleratorResources = ['device2'];
+
+      // THEN
+      expect(() => {
+        taskDefinition.addContainer('cont', {
+          image: ecs.ContainerImage.fromRegistry('test'),
+          memoryLimitMiB: 1024,
+          inferenceAcceleratorResources,
+        });
+      }).toThrow();
```

Comment:
Can we be more specific about the error message like https://github.com/aws/aws-cdk/blob/bbcbee652214b40198d2a0b540b1f2cf6c0bef2e/packages/%40aws-cdk/aws-ecs/test/container-definition.test.ts#L242